nd4j/nd4j-backends/nd4j-backend-impls/nd4j-cuda/src/main/java/org/nd4j/linalg/jcublas/buffer/BaseCudaDataBuffer.java

```diff
@@ -907,26 +928,179 @@ public void set(float[] data, long length, long srcOffset, long dstOffset) {
      * @param srcOffset
      * @param dstOffset
      */
-    public void set(double[] data, long length, long srcOffset, long dstOffset) {
+    public void set(short[] data, long length, long srcOffset, long dstOffset) {
         switch (dataType()) {
             case BOOL:  {
                 val pointer = new BytePointer(ArrayUtil.toBytes(data));
                 copyDataFromSrc(pointer,length,offset,dstOffset);
 
             }
             break;
+            case UBYTE: {
+                data = ArrayUtil.cutBelowZero(data);
+            }
```

Comment:
Is this explicitly meant to fall through to the next case? If so, put a note here.